### DISCUSSION TITLE
How about extract the monitor part out of HA custom-components?

### DISCUSSION THREAD

**Discussion by aheadlead** [custom-components/ble_monitor#370] [Ideas]
Hi Ernst,

Thanks for your great efforts on this project and I really appreciate your work.

I have a idea that any possibility to extract the parser/monitor part out of HA custom-components? To make that part as a dedicated python library as python-miio and refer(or import) this library in current custom components? If so, I can simply build some toys to monitor my BLE sensor or remote without a heavy HA installation.

Also I have a question. My house is big enough that I can not cover my whole house by only one bluetooth dougle. Is there any existing solution to solve it? One idea I have is to buy several raspberrypis (with bluetooth version), put them all over my house and make it to be a BLE agent of your project ble_monitor (maybe over gRPC?).

Thanks,
Yulan

**Comment by Ernst79** [comment]
Hi,

I'm slowly moving towards the possibility of getting the parser out of BLE monitor, but don't expect that to be done soon. There are more people asking for this, I know, but my time is limited and I'm not a programmer, just doing this as a hobby. 

I've started to move the parser towards a different folder (ble_parser), which will make it easier later to get things to a pypi package. aioblescan also has some ATC support already, so another option might be to move things to aioblescan pypi package. But first I want to rewrite the parser a bit to make it easier to maintain. 

You next question, ESPhome is the way to go for that. You can bye a cheap ESP chip for a few Euro/Dollar/... and receive data from quite some BLE sensors. Not all sensors that are supported by BLE monitor are available though.
